### ISSUE TITLE
Far more buffering than plain youtube

### ISSUE THREAD

**Issue by seth-scherer** [TeamPiped/Piped#278]
Constant buffering / videos not loading at all seems to be a consistent issue, that I face only while using Piped as opposed to vanilla Youtube

**Comment by FireMasterK** [comment]
You could try setting a default quality and increasing the default quality, this should help significantly.

The US server is also significantly slower, for some odd reason, it has now been temporarily disabled.

**Comment by seth-scherer** [comment]
> You could try setting a default quality and increasing the default quality, this should help significantly.
> 
> The US server is also significantly slower, for some odd reason, it has now been temporarily disabled.

I have a default quality set

**Comment by FireMasterK** [comment]
Do you still have this problem?

**Comment by seth-scherer** [comment]
> Do you still have this problem?

yes, i've had it long before i set a default quality

**Comment by FireMasterK** [comment]
Could you send a screenshot of your timings/waterfall graph in the network tab of your browser's web tools?

Example screenshot:
![image](https://user-images.githubusercontent.com/20838718/126213366-7ba0da59-4856-43d9-88bf-c402eb51bb92.png)

**Comment by seth-scherer** [comment]
Well actually, I switched to 720p and I haven't had any issues. Which is strange because I don't have any buffering issues with 1080p on youtube

**Comment by seth-scherer** [comment]
> Could you send a screenshot of your timings/waterfall graph in the network tab of your browser's web tools?
> 
> Example screenshot:
> ![image](https://user-images.githubusercontent.com/20838718/126213366-7ba0da59-4856-43d9-88bf-c402eb51bb92.png)

![2021-07-20-115515_265x545_scrot](https://user-images.githubusercontent.com/86060246/126356248-37a89c6c-9288-468b-909e-1285308233f2.png)

**Comment by FireMasterK** [comment]
Since you're using Firefox, could you try disabling `network.http.http3.enabled` in `about:config` temporarily and test again?

**Comment by seth-scherer** [comment]
> Since you're using Firefox, could you try disabling `network.http.http3.enabled` in `about:config` temporarily and test again?
> 
> Firefox had a slow implementation of http3 the last time I checked it.

it's actually already off, maybe because i'm using LibreWolf.
I'm having considerably less issues with 720p set as my default quality, so I might just stick with this tbh

**Comment by moriel5** [comment]
I had mentioned that I have this issue as well in issue #224 (in a slightly different context), and I don't think that you do not have an issue with 720p, but rather that the bandwidth needed for 720p is much smaller.

And @FireMasterK, while I could very well be wrong, I do not think that this has to do with Firefox's HTTP3 implementation, as my tests showed the same results on both Firefox and Edge, on both Windows and Linux, with HTTP3 enabled and disabled.

**Comment by seth-scherer** [comment]
> I had mentioned that I have this issue as well in issue #224 (in a slightly different context), and I don't think that you do not have an issue with 720p, but rather that the bandwidth needed for 720p is much smaller.
> 
> And @FireMasterK, while I could very well be wrong, I do not think that this has to do with Firefox's HTTP3 implementation, as my tests showed the same results on both Firefox and Edge, on both Windows and Linux, with HTTP3 support enabled and disabled.

Yeah you're right, but it seems to be even worse now, at 720p I'm getting constant buffering. It's making Piped nearly unusable sometimes.

**Comment by FireMasterK** [comment]
A major change has been made, could you tell if it is better?

With these changes, even with a VPN to NL, I'm able to watch 1080p content at a good speed.

**Comment by moriel5** [comment]
In terms of buffering, I'm either getting the same or slightly better (still testing to make sure that this is not placebo), however Firefox isn't offloading anything to the GPU, for some odd reason (will be posting an issue at the Solus Phabricator (Solus's code hub)), and interestingly, Piped seems to require more CPU time than Invidious or CloudTube (other system resources are perfectly fine, and just as always).

**Comment by moriel5** [comment]
I just tested, and am getting... weird results.

For reference, all tests in this post were done on a Lenovo IdeaPad 300-15ISK, with an i5-6200U and a Radeon R5 M330 (Lenovo's stock base clock is 750Mhz instead of 955Mhz) (18W TGP) and 8GB (4+4) of 1600Mhz DDR3L RAM and a 7200RPM SATA HDD.

I disabled SponsorBlock, set the light theme, disabled autoplay (which keeps on reenabling itself whenever some of the other some are changed, will be a separate issue once I have the time to reproduce the details), set default resolution to 720p and kept the buffering goal at 10 seconds. I made sure these settings were set for each test.

Only CPU load was tested, not any other external variable.

Video tested was https://piped.kavin.rocks/watch?v=5_LOB-_WhJI

720p
AVC: ~%40-60 (mostly ~%44-50)
VP9: ~%43-71 (mostly ~%44-54)
AV1: ~%41-77 (mostly ~%57-69)
All selected: ~%44-68 (mostly ~%52-68)

1080p
AVC: ~%48-71 (mostly ~%48-64)
VP9: ~%32-74 (mostly ~%44-65)
AV1: ~%43-84 (mostly ~%49-67)
All selected: ~%43-80 (mostly ~%44-70)

**Comment by FireMasterK** [comment]
> 
> 720p
> AVC: ~%40-60 (mostly ~%44-50)
> VP9: ~%43-71 (mostly ~%44-54)
> AV1: ~%41-77 (mostly ~%57-69)
> All selected: ~%44-68 (mostly ~%52-68)
> 
> 1080p
> AVC: ~%48-71 (mostly ~%48-64)
> VP9: ~%32-74 (mostly ~%44-65)
> AV1: ~%43-84 (mostly ~%49-67)
> All selected: ~%43-80 (mostly ~%44-70)

Something doesn't add up here.

Enabling only AV1 or All selected will default to VP9 since it's the lowest bandwidth.

**Comment by moriel5** [comment]
@FireMasterK I agree, especially since these are very rough averages since it was wildly inconsistent (I also had a few disconnected cases where I had CPU utilization in the 20s, only for it to jump up to the 80s, regardless of the resolution and codec).
